shell/artifacts/Common/source/Multiplexer.js

```diff
@@ -31,7 +31,7 @@ defineParticle(({TransformationDomParticle}) => {
           continue;
         }
         otherMappedViews.push(`map '${await arc.mapHandle(otherView._proxy)}' as v${index}`);
-        let hostedOtherConnection = hostedParticle.connections.find(conn => conn.type.equals(otherView.type));
+        let hostedOtherConnection = hostedParticle.connections.find(conn => otherView.type.equalsOrSubclass(conn.type));
```

Comment:
This should use the type checker to determine whether the types are compatible, like this: https://github.com/PolymerLabs/arcs/blob/master/runtime/recipe/handle-connection.js#L115

Reply:
Added a helper method to connection-spec, using the TypeChecker.compareTypes.
PTAL